### DISCUSSION TITLE
v0.10.2

### DISCUSSION THREAD

**Discussion by laipz8200** [langgenius/dify#9956] [Releases]
## ✨ What’s New in v0.10.2? ✨

We're back with version 0.10.2, packed with new features, crucial fixes, and improvements that will make your workflow smoother and more efficient. Let's break down what's included:

### 🚀 New Features

- **Markdown Buttons in Workflows:** Sprinkle some buttons in your markdown for workflows to make them pop. Shout out to @crazywoola in #9876.

- **Document Extractor Node with JSON Support:** You can now manage JSON types with extractor nodes for more flexible data extraction. Thanks to @Theysua in #9899.

- **Text-to-Image with Stable-Diffusion 3.5:** Bring your text to life with the new Stable-Diffusion-3.5-large integration using siliconflow. Props to @hwzhuhao in #9909.

- **Vectorizer Support in Workflows:** The Vectorizer tool is now at your disposal within workflows for better data manipulation—kudos to @hjlarry in #9932.

### 🛠️ Enhancements and Fixes

- **API Documentation Upgrade:** We've polished our API documentation for advanced chat and workflow functionalities. Special thanks to @kurokobo in #9882.

- **LLM Node Streaming Output Fix:** Fixed the LLM Node streaming output to properly function after IF-ELSE decisions, ensuring your workflows are on track. Thanks to @laipz8200 in #9875.

- **Workflow Code Generator Update:** Cleared up the Generate button from incorrect locations for a more intuitive workflow setup. Update by @Kota-Yamaguchi in #9740.

- **Enhanced Prompt Handling in Workflows:** Added vision support in prompts for more dynamic configurations. Nicely done by @laipz8200 in #9790.

- **Improved File Handling in HTTP Requests:** Your HTTP requests within workflows are now more robust. Much appreciated, @laipz8200 in #9944.

- **⚠️ Fix for Tool Var Type Error:** If you're using a "Number" parameter in "Workflow as Tool," please remove and re-add this tool to avoid any issues. Thanks to @iamjoel in #9937.

- **Chat Log Display Fix:** We've resolved the display problems for chat logs, making sure everything shows up correctly. Big thanks to @xuzuodong in #9777.

- **Vision Support for Tool Image Gen:** Image generation with vision models is now supported within your workflows—credit to @hjlarry in #9871.

- **System Parameters Standardization:** Our system parameters have been standardized across the board. Cheers to @laipz8200 in #9797.

- **Image Previews in DOCX:** Fixed issues with previews in 'docx' knowledge bases. Thanks again to @laipz8200 in #9893.

This release is all about enhancing your workflow capabilities and providing robust tools and documentation to back them up. Keep the feedback coming and happy building! 🌟

---

### Docker compose deployments

1. Back up your customized docker-compose YAML file (optional)

   ```bash
   cd docker
   cp docker-compose.yaml docker-compose.yaml.$(date +%s).bak
   ```

2. Get the latest code from the main branch

   ```bash
   git checkout main
   git pull origin main
   ```

3. Stop the service，Command, please execute in the docker directory

   ```bash
   docker compose down
   ```

4. Back up data

   ```bash
   tar -cvf volumes-$(date +%s).tgz volumes
   ```

5. Upgrade services

   ```bash
   docker compose up -d
   ```

### Source Code deployments

1. Stop API server, Worker and Web frontend Server.

2. Get the latest code from the release branch:

   ```bash
   git checkout 0.10.2
   ```

3. Update Python dependencies:

   ```bash
   cd api
   poetry install
   ```

7. Then, let's run the migration script:

   ```bash
   poetry run flask db upgrade
   ```

8. Finally, run API server, Worker and Web frontend Server again.

---

## What's Changed
* Update Code Generator to use the currently configured model. by @Kota-Yamaguchi in https://github.com/langgenius/dify/pull/9740
* fix(api): handle missing upload_file_id for tool_file messages by @laipz8200 in https://github.com/langgenius/dify/pull/9756
* feat: regenerate history switch navigation by @xuzuodong in https://github.com/langgenius/dify/pull/8749
* feat:Add host volume env variables for postgres, redis and weaviate by @jter in https://github.com/langgenius/dify/pull/9761
* fix: fe can not start by @iamjoel in https://github.com/langgenius/dify/pull/9768
* Update README_CN.md by @sunner in https://github.com/langgenius/dify/pull/9766
* fix: chat log not showing correctly by @xuzuodong in https://github.com/langgenius/dify/pull/9777
* refactor(file_factory): improve filename and mime type determination by @laipz8200 in https://github.com/langgenius/dify/pull/9784
* fix(workflow): enhance prompt handling with vision support by @laipz8200 in https://github.com/langgenius/dify/pull/9790
* remove unstructured pdf extract by @JohnJyong in https://github.com/langgenius/dify/pull/9794
* fix: chart tool can't display chinese by @hjlarry in https://github.com/langgenius/dify/pull/9686
* chore(docker): correct package version for expat and perl in Dockerfile by @laipz8200 in https://github.com/langgenius/dify/pull/9801
* openai compatiable api usage and id by @guogeer in https://github.com/langgenius/dify/pull/9800
* chore(docker): add default for MAX_VARIABLE_SIZE in docker-compose by @laipz8200 in https://github.com/langgenius/dify/pull/9798
* feat(parameters): standardize system parameter field types and values by @laipz8200 in https://github.com/langgenius/dify/pull/9797
* chore: abstract common function with local storage by @yaoice in https://github.com/langgenius/dify/pull/9811
* Added description for .ppt, specify the reason for unstructured.io by @Theysua in https://github.com/langgenius/dify/pull/9452
* chore: make doc extractor node also can extract text by file extension by @hjlarry in https://github.com/langgenius/dify/pull/9543
* Feat: use file size limit from api by @JzoNgKVO in https://github.com/langgenius/dify/pull/9739
* feat(api): add generic file size limit parameter by @laipz8200 in https://github.com/langgenius/dify/pull/9812
* refactor: update load_stream method to directly yield file chunks by @hwzhuhao in https://github.com/langgenius/dify/pull/9806
* fix: duckduckgo image search not work by @hjlarry in https://github.com/langgenius/dify/pull/9821
* feat: create_empty_dataset api add the description parameter and update api docs by @rchlz in https://github.com/langgenius/dify/pull/9824
* fix: yuque book id should be string by @crazywoola in https://github.com/langgenius/dify/pull/9819
* chore: add local storage test by @yaoice in https://github.com/langgenius/dify/pull/9827
* add tidb on qdrant type by @JohnJyong in https://github.com/langgenius/dify/pull/9831
* fix: exclude failed answer when sending messages by @xuzuodong in https://github.com/langgenius/dify/pull/9835
* fix: moonshot response_format raise error by @hjlarry in https://github.com/langgenius/dify/pull/9847
* chore: code generator button should only display in code node by @hjlarry in https://github.com/langgenius/dify/pull/9842
* Fix: add check for maximum chunk length by @JzoNgKVO in https://github.com/langgenius/dify/pull/9837
* chore: update api docs by @JzoNgKVO in https://github.com/langgenius/dify/pull/9832
* chore: translate i18n files by @github-actions in https://github.com/langgenius/dify/pull/9853
* Marking the last piece of data on each page is a duplicate issue, which can be solved by adding the id field to the order by rig and using a unique field by @yuanboao in https://github.com/langgenius/dify/pull/9799
* feat: allow answer node use chat_var and env_var by @hjlarry in https://github.com/langgenius/dify/pull/9226
* fix: segement settings of documents raise error by @hjlarry in https://github.com/langgenius/dify/pull/8971
* upgrade nltk, unstructured and starlette by @JohnJyong in https://github.com/langgenius/dify/pull/9860
* chore: add default value for redis configuration by @hwzhuhao in https://github.com/langgenius/dify/pull/9864
* feat: support comfyui workflow tool image generate image by @hjlarry in https://github.com/langgenius/dify/pull/9871
* feat: support button in markdown by @crazywoola in https://github.com/langgenius/dify/pull/9876
* fix: wrong element object by @virgosoy in https://github.com/langgenius/dify/pull/9868
* fix(workflow): Take back LLM streaming output after IF-ELSE by @laipz8200 in https://github.com/langgenius/dify/pull/9875
* fix users had already joined a workspace, but the system still first … by @G81192 in https://github.com/langgenius/dify/pull/9834
* docs: improve api documentation for advanced chat and workflow by @kurokobo in https://github.com/langgenius/dify/pull/9882
* Update README.md by @leilei-jiang in https://github.com/langgenius/dify/pull/9886
* Update README.md by @leilei-jiang in https://github.com/langgenius/dify/pull/9889
* Update README.md by @leilei-jiang in https://github.com/langgenius/dify/pull/9890
* Update README.md by @leilei-jiang in https://github.com/langgenius/dify/pull/9891
* fix(api): add signature generation for image previews by @laipz8200 in https://github.com/langgenius/dify/pull/9893
* feat(workflow): Support JSON type in document extractor node by @Theysua in https://github.com/langgenius/dify/pull/9899
* chore: Extract common functions of the base model in Azure OpenAI Provider by @yaoice in https://github.com/langgenius/dify/pull/9907
* fix(tools): remove the undefined variable parameter_type by @hwzhuhao in https://github.com/langgenius/dify/pull/9908
* feat: add stable-diffusion-3-5-large for the text-to-image tool with siliconflow by @hwzhuhao in https://github.com/langgenius/dify/pull/9909
* fix: enterprise create workspace by @ZhouhaoJiang in https://github.com/langgenius/dify/pull/9921
* add document lock for multi-thread by @JohnJyong in https://github.com/langgenius/dify/pull/9873
* Remove invalid languages error by @seikyo-cho-lvgs in https://github.com/langgenius/dify/pull/9928
* fix: tool var type error by @iamjoel in https://github.com/langgenius/dify/pull/9937
* Modify and add jp translation by @HanqingZ in https://github.com/langgenius/dify/pull/9930
* feat: add models for gitee.ai by @fchange in https://github.com/langgenius/dify/pull/9490
* feat: support Vectorizer can be used in workflow by @hjlarry in https://github.com/langgenius/dify/pull/9932
* feat(tools): add Baidu translation tool by @XiaoLey in https://github.com/langgenius/dify/pull/9943
* feat(http-request-executor): enhance file handling in HTTP requests by @laipz8200 in https://github.com/langgenius/dify/pull/9944
* fix: separator change add too many backslash by @iamjoel in https://github.com/langgenius/dify/pull/9949
* Docs: fix docs url by @AllenWriter in https://github.com/langgenius/dify/pull/9954
* Fix/external api update by @JohnJyong in https://github.com/langgenius/dify/pull/9955
* chore: update version to 0.10.2 in packaging and docker configurations by @laipz8200 in https://github.com/langgenius/dify/pull/9924

## New Contributors
* @jter made their first contribution in https://github.com/langgenius/dify/pull/9761
* @sunner made their first contribution in https://github.com/langgenius/dify/pull/9766
* @rchlz made their first contribution in https://github.com/langgenius/dify/pull/9824
* @yuanboao made their first contribution in https://github.com/langgenius/dify/pull/9799
* @virgosoy made their first contribution in https://github.com/langgenius/dify/pull/9868
* @G81192 made their first contribution in https://github.com/langgenius/dify/pull/9834
* @seikyo-cho-lvgs made their first contribution in https://github.com/langgenius/dify/pull/9928

**Full Changelog**: https://github.com/langgenius/dify/compare/0.10.1...0.10.2

<hr /><em>This discussion was created from the release <a href='https://github.com/langgenius/dify/releases/tag/0.10.2'>v0.10.2</a>.</em>

**Comment by rshan1515** [comment]
Hola help me out

**Reply by iamjoel** [reply]
Is there an issue you'd like to discuss?
